payments-core/src/main/java/com/stripe/android/payments/paymentlauncher/PaymentLauncherViewModel.kt

```diff
@@ -77,48 +85,70 @@ internal class PaymentLauncherViewModel @Inject constructor(
+
     /**
      * [PaymentResult] live data to be observed.
      */
     internal val paymentLauncherResult = MutableLiveData<PaymentResult>()
 
+    /**
+     * Registers the calling activity to listen to payment flow results. Should be called in the
+     * activity onCreate.
+     */
+    internal fun register(caller: ActivityResultCaller, host: AuthActivityStarterHost) {
+        authActivityStarterHost = host
+        authenticatorRegistry.onNewActivityResultCaller(
+            caller,
+            ::onPaymentFlowResult
+        )
+    }
+
     /**
      * Confirms a payment intent or setup intent
      */
-    internal suspend fun confirmStripeIntent(confirmStripeIntentParams: ConfirmStripeIntentParams) {
-        savedStateHandle.set(KEY_HAS_STARTED, true)
-        logReturnUrl(confirmStripeIntentParams.returnUrl)
-        val returnUrl =
-            if (isInstantApp) {
-                confirmStripeIntentParams.returnUrl
-            } else {
-                confirmStripeIntentParams.returnUrl.takeUnless { it.isNullOrBlank() }
-                    ?: defaultReturnUrl.value
-            }
-        runCatching {
-            confirmIntent(confirmStripeIntentParams, returnUrl)
-        }.fold(
-            onSuccess = { intent ->
-                intent.nextActionData?.let {
-                    if (it is StripeIntent.NextActionData.SdkData.Use3DS1) {
-                        intent.id?.let { intentId ->
-                            threeDs1IntentReturnUrlMap[intentId] = returnUrl.orEmpty()
+    @VisibleForTesting
+    internal fun confirmStripeIntent(confirmStripeIntentParams: ConfirmStripeIntentParams) {
+        viewModelScope.launch {
+            savedStateHandle.set(KEY_HAS_STARTED, true)
+            logReturnUrl(confirmStripeIntentParams.returnUrl)
+            val returnUrl =
+                if (isInstantApp) {
+                    confirmStripeIntentParams.returnUrl
+                } else {
+                    confirmStripeIntentParams.returnUrl.takeUnless { it.isNullOrBlank() }
+                        ?: defaultReturnUrl.value
+                }
+            runCatching {
+                confirmIntent(confirmStripeIntentParams, returnUrl)
+            }.fold(
+                onSuccess = { intent ->
+                    intent.nextActionData?.let {
+                        if (it is StripeIntent.NextActionData.SdkData.Use3DS1) {
+                            intent.id?.let { intentId ->
+                                threeDs1IntentReturnUrlMap[intentId] = returnUrl.orEmpty()
+                            }
                         }
                     }
+                    authActivityStarterHost?.let {
+                        authenticatorRegistry.getAuthenticator(intent).authenticate(
+                            it,
+                            intent,
+                            apiRequestOptionsProvider.get()
+                        )
+                    }
+                },
```

Comment:
looks like we're keeping a reference to `authActivityStarterHost` just to be able to run this logic right? 

We're unlocking rotations with this PR, meaning every time the user rotates we're stacking an activity in memory that cannot be GCed, eventually running into an OOM / crash. 

An important follow-up of this would be:

- Move `authActivityStarterHost` to activity
- Move this logic to the activity Communicate from viewModel to activity via a `SharedFlow` or [LiveData<Event<N>>](https://github.com/stripe/stripe-android/blob/957b89658c9873e9c98e37f11223add5fc700b84/paymentsheet/src/main/java/com/stripe/android/paymentsheet/viewmodels/BaseSheetViewModel.kt#L340)

Reply:
+1 to this. I'm not too thrilled about holding on to this `AuthActivityStarterHost` in the view model. I think flow is the way to go. We've been using those over LiveData.